### PR TITLE
refactor add_value_features and reindex using tables instead of dicts

### PR DESCRIPTION
**Pull request by songhappy**
move add_value_features to scala side to boost performance.
refactor add_value_features  and reindex to take tables instead of python dicts.
refactor gen_reindex_mapping  to output tables instead of python dicts.

### CODE REVIEW

**Comment by songhappy** [comment]
jenkins passed, the abortion is irrelevant.
http://10.112.231.51:18888/job/ZOO-PR-Validation/6454/

File: zoo/src/main/scala/com/intel/analytics/zoo/friesian/python/PythonFriesian.scala [hunk 1]
```diff
@@ -288,6 +288,47 @@ class PythonFriesian[T: ClassTag](implicit ev: TensorNumeric[T]) extends PythonZ
...
+
+    var tmpDF = df
+    for(col <- cols.asScala.toList) {
+      tmpDF = addValueSingleCol(tmpDF, col, mapBr, key, value)
+    }
+
+    tmpDF
+  }
+
+  def addValueSingleCol(df: DataFrame, colName: String, mapBr: Broadcast[Map[Int, Int]],
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/friesian/python/PythonFriesian.scala:306]: move to `features.Utils`

**Comment by songhappy** [comment]
jenkins, the abortion is irrelevant.  
http://10.112.231.51:18888/view/ZOO-PR/job/ZOO-PR-Validation/6488/

File: zoo/src/main/scala/com/intel/analytics/zoo/friesian/python/PythonFriesian.scala (hunk 1 above)
Comment (songhappy) [reply zoo/src/main/scala/com/intel/analytics/zoo/friesian/python/PythonFriesian.scala]: done

**Pull request merged**